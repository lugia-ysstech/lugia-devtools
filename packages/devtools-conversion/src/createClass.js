/**
 *
 * create by guorg
 *
 * @flow
 */
import { camelNamed, recursiveChildren } from './utils';
import { widgetIdInLugiax } from './widgetIdInLugiax';
import { createThemeCode } from './theme';

export function createExpantWidgetid(target: ?(Object[])): string[] {
  const result = [];
  if (!target || target.length < 1) {
    return result;
  }
  recursiveChildren(target, result);

  return result;
}

export function createCustomLabel(
  widgetName: string,
  componentName: string
): string {
  let code = '';
  if (!widgetName || !componentName) {
    return code;
  }
  code =
    'const ' +
    componentName +
    ' = styled.' +
    widgetName +
    '`' +
    'font-size: 16px;' +
    '`;';
  return code;
}

export function createClassCode(
  id: string,
  childPadInfo: Object,
  lugiax: Object,
  themes: Object,
  index: number | string,
  widgetId2Component: Object,
  isResponsive: boolean,
  options: Object
): string {
  let classCode = '';
  const padInfo = childPadInfo[id];
  if (padInfo) {
    const { layers, id2WidgetInfo } = padInfo;
    const { layerCode, layerBindCode } = createLayerComponent(
      layers,
      id2WidgetInfo,
      lugiax,
      themes,
      index,
      widgetId2Component,
      isResponsive,
      options
    );
    classCode = `${layerBindCode} class ChildPadComponent${index} extends React.Component {
      render(){
        return (
          <React.Fragment>
            ${layerCode}
          </React.Fragment>
        );
      }
    };`;
  }
  return classCode;
}

export function createLayerComponent(
  layers: Object[],
  id2WidgetInfo: Object,
  lugiax: Object,
  themes: Object,
  index: number | string,
  widgetId2Component: Object,
  isResponsive: boolean,
  options: Object
): Object {
  const { widgetIdHasAssetPropsName = {}, resourcesHead = '', images = [] } = options || {};
  let layerCode = '',
    layerBindCode = '';
  layers.forEach((key: Object, i: number) => {
    const { id: layerId, width, height, zIndex, point } = key;
    const layerInfo = id2WidgetInfo[layerId];
    const { module, widgetName, props } = layerInfo;
    const componentName = camelNamed(widgetName);
    // if (module === '@lugia/lugia-web-html') {
    //   const customLabelName = camelNamed(`${widgetName}${index}${i}`);
    //   widgetId2Component[layerId] = customLabelName;
    //   customLabelCode = customLabelCode + createCustomLabel(widgetName, customLabelName);
    // }
    // const name = widgetId2Component[layerId]
    //   ? widgetId2Component[layerId]
    //   : widgetName;
    let name = widgetId2Component[layerId] || widgetName;
    if (module === '@lugia/lugia-web-html') {
      name = widgetId2Component[widgetName];
    }
    const newIndex = index.toString() + i;
    const lugiaxInfo = widgetIdInLugiax(layerId, lugiax, name, newIndex);
    if (lugiaxInfo) {
      const { inLugiax, lugiaxCode, componentName: bindName } = lugiaxInfo;
      widgetId2Component[layerId] = inLugiax ? bindName : componentName;
      if (inLugiax) {
        layerBindCode = layerBindCode + lugiaxCode;
      }
    }
    const hasContainer = 'TargetContainer' in props;
    const assetProps = widgetIdHasAssetPropsName[layerId];
    const { containerStart, containerEnd } = createTargetContainer(
      props.TargetContainer,
      props,
      { assetProps, resourcesHead, images }
    );
    const containerStartLabel = hasContainer ? containerStart : '';
    const containerEndLabel = hasContainer ? containerEnd : '';
    const propsConfig = getComponentProps(props, { assetProps, resourcesHead, images });

    const layerThemeInfo =
      themes && themes.widgetId2ThemeInfo && themes.widgetId2ThemeInfo[layerId];
    const { viewClass, configString } = createThemeCode(
      themes.widgetId2ThemeInfo,
      layerId
    );
    const viewClassCode = layerThemeInfo
      ? ' viewClass="' + viewClass + '"'
      : '';
    const containerThemeCode = containerStartLabel
      ? containerStartLabel + viewClassCode + '>'
      : containerStartLabel;
    const componentThemeCode = hasContainer ? '' : viewClassCode;
    const commonStr = `context.getLayout("${layerId}").`;
    const styleWidth = isResponsive ? `${commonStr}width` : width;
    const styleHeight = isResponsive ? `${commonStr}height` : height;
    const styleLeft = isResponsive ? `${commonStr}point[0]` : point[0];
    const styleRight = isResponsive ? `${commonStr}point[1]` : point[1];
    const theContext = isResponsive ? 'context' : 'undefined';
    const themeStr = configString ? `${configString},` : '{}';
    layerCode =
      layerCode +
      `<div style={{position: 'absolute',width: ${styleWidth} + 'px',
        height: ${styleHeight} + 'px', zIndex: '${zIndex}', 
        left: ${styleLeft} + 'px', 
        top: ${styleRight} + 'px' }}
        ><Theme config={{'${layerId}':themeHandle('${layerId}',${theContext}, ${themeStr})}}>${containerThemeCode}<${widgetId2Component[layerId]} ${componentThemeCode} ${propsConfig} />${containerEndLabel}</Theme></div>`;
  });

  if (layerCode && isResponsive) {
    layerCode = `<ResponsiveContext.Consumer>{
                    context => {
                        return <React.Fragment>${layerCode}</React.Fragment>
                    }
                }</ResponsiveContext.Consumer>`;
  }

  return { layerCode, layerBindCode };
}

export function getComponentProps(props: ?Object, opt: ?Object): string {
  if (!props) {
    return '';
  }
  const comProps = [];
  const { assetProps = {}, resourcesHead = '', images = [] } = opt || {};
  const propsKey = Object.keys(props);
  if (propsKey && propsKey.length > 0) {
    propsKey.forEach((item: string) => {
      if (item !== 'TargetContainer') {
        const propsItem = props && props[item];
        const assetPropItem = assetProps[item];
        if (assetPropItem && typeof propsItem === 'string') {
          const url = `${resourcesHead}${propsItem}`;
          const propsValue = `image${images.length}`;
          images.push({
            name: propsValue,
            value: url,
          });
          comProps.push(`${item}={ ${propsValue} }`);
        } else {
          comProps.push(`${item}=${handlePropsType(propsItem)}`);
        }

      }
    });
  }

  return comProps.join(' ');
}
export function createTargetContainer(
  TargetContainer: ?Object,
  props: Object,
  opt: Object
): Object {
  let labelStart = '',
    labelEnd = '';
  if (!TargetContainer) {
    return { containerStart: labelStart, containerEnd: labelEnd };
  }
  const { widgetName } = TargetContainer;
  const propsCode = getComponentProps(props, opt);
  labelStart = `<${widgetName} ${propsCode}`;
  labelEnd = `</${widgetName}>`;

  return { containerStart: labelStart, containerEnd: labelEnd };
}

export function handlePropsType(propsItem?: any): string {
  if (!propsItem) {
    return '""';
  }
  const type = typeof propsItem;
  if (type === 'object') {
    return `{ ${JSON.stringify(propsItem)} }`;
  }
  if (type === 'number') {
    return `{ ${propsItem} }`;
  }
  if (type === 'string') {
    return `{  \`${propsItem}\` }`;
  }
  return `{'${propsItem}'}`;
}

export function createComponent(
  childrenPad: ?(Object[]),
  widgetId2ChildPadInfo: Object,
  lugiax: Object,
  themes: Object,
  widgetId2Component: Object,
  isResponsive: boolean,
  options: Object
): string {
  let classCode = '';
  if (!childrenPad || childrenPad.length < 1) {
    return classCode;
  }
  const padWidgetIds = createExpantWidgetid(childrenPad);
  padWidgetIds.forEach((id: string, index: number) => {
    const componentName = `ChildPadComponent${index}`;
    classCode =
      classCode +
      createClassCode(
        id,
        widgetId2ChildPadInfo,
        lugiax,
        themes,
        index,
        widgetId2Component,
        isResponsive,
        options
      );
    widgetId2Component[id] = componentName;
    const lugiaxInfo = widgetIdInLugiax(id, lugiax, componentName, index);
    if (lugiaxInfo) {
      const { inLugiax, lugiaxCode, componentName: bindName } = lugiaxInfo;
      widgetId2Component[id] = inLugiax ? bindName : componentName;
      const bindCode = inLugiax ? lugiaxCode : '';
      classCode = classCode + bindCode;
    }
  });

  return classCode;
}
