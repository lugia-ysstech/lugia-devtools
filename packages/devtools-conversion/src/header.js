/**
 *
 * create by guorg
 *
 * @flow
 */
import { camelNamed } from './utils';
import type { HeaderType } from '@lugia/devtools-conversion';

export function createHeader(
  mainDependencies: ?(Object[]),
  widgetId2Component: Object
): HeaderType {
  let packages =
    'import React from "react";' +
    'import styled from "styled-components";' +
    'import { Theme } from "@lugia/lugia-web";';
  let styledComponentCode = '';
  const moudleMap = {};

  if (!mainDependencies || mainDependencies.length < 1) {
    return { packages, styledComponentCode };
  }
  mainDependencies.forEach((item: Object) => {
    const { widgetName, module } = item;
    const componentName = camelNamed(widgetName);
    if (module === '@lugia/lugia-web-html') {
      styledComponentCode =
        'const ' +
        componentName +
        ' = styled.div' +
        '`' +
        'font-size: 12px;' +
        '`;';
      widgetId2Component[widgetName] = componentName;
    } else {
      const components = widgetName.split('.');
      if (!moudleMap[components[0]]) {
        packages += `import { ${components[0]} } from '${module}';`;
      }
      moudleMap[components[0]] = true;
    }
  });
  return { packages, styledComponentCode };
}
export function getModelCode(lugiax: ?Object): string {
  let code = '';
  if (!lugiax) {
    return code;
  }
  const {
    widgetId2PropsName2BindInfo = {},
    widgetId2EventName2MutationInfo = {},
    page2mutation = {},
  } = lugiax;

  const propsModel = getTargetModal(widgetId2PropsName2BindInfo);
  const eventModal = getTargetModal(widgetId2EventName2MutationInfo);
  const pageModal = [];
  const pageMutationsKey = Object.keys(page2mutation);
  for (let i = 0; i < pageMutationsKey.length; i++) {
    const { modelName } = page2mutation[pageMutationsKey[i]];
    if (modelName) {
      pageModal.push(modelName);
    }
  }
  let modelName = [ ...propsModel, ...eventModal, ...pageModal ];
  if (modelName.length > 0) {
    modelName = [ ...new Set(modelName) ];
    modelName.forEach((model: string) => {
      code = code + `import ${model} from '../models/${model}';`;
    });
  }
  return code;
}
export function getTargetModal(target: ?Object): any[] {
  const keys = [];
  if (!target) {
    return keys;
  }
  const targetKeys = Object.keys(target);
  if (!targetKeys || targetKeys.length < 1) {
    return keys;
  }
  targetKeys.forEach((targetItem: string) => {
    const targetValue = target && target[targetItem];
    const value = Object.values(targetValue);
    value.forEach((item: Object) => {
      item && item.modelName && keys.push(item.modelName);
    });
  });
  return keys;
}

export function getLugiaDCoreCode(
  lugiax: Object
): string[] {
  if (!lugiax) {
    return '';
  }
  const functionNames = [ 'themeHandle' ];
  let lugiaxCode = '';
  const { widgetId2PropsName2BindInfo = {}, widgetId2EventName2MutationInfo = {} } = lugiax || {};
  const bindKeys = Object.keys(widgetId2PropsName2BindInfo);
  const mutationKeys = Object.keys(widgetId2EventName2MutationInfo);
  if (bindKeys.length || mutationKeys.length) {
    functionNames.push('bindHandleEvent');
    functionNames.push('getData');
    lugiaxCode = 'import lugiax, { bindTo, connect, bind } from \'@lugia/lugiax\';';
  }

  return lugiaxCode + createLugiaDCoreCode(functionNames);
}

export function createLugiaDCoreCode(names: string[]): string {
  if (!names || !names.length) {
    return '';
  }

  const functionNames = names.join(',');
  return `import {${functionNames}} from '@lugia/lugia-web/dist/common/lugiadCore';`;
}

export function hasResponsive(layoutInfos: Object): boolean {
  const { mode2Config = {}, mode2LayoutData = {} } = layoutInfos || {};
  const themeaMainPadKeys = Object.keys(mode2Config);
  const themeComponentKeys = Object.keys(mode2LayoutData);
  return themeaMainPadKeys.length > 1 || themeComponentKeys.length > 1;
}

export function getResponsiveCode(layoutInfos: Object): Object {
  let rspPackagesCode = '',
    rspDeconstructionCode = '';
  if (!layoutInfos) {
    return {
      rspPackagesCode,
      rspDeconstructionCode,
    };
  }
  if (hasResponsive(layoutInfos)) {
    rspPackagesCode = 'import { DesignResponsive } from \'@lugia/lugia-web\';';
    rspDeconstructionCode = 'const ResponsiveContext = DesignResponsive.ResponsiveContext;';
  }

  return {
    rspPackagesCode,
    rspDeconstructionCode,
  };
}
