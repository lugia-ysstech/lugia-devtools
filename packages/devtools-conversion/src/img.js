/*
 *
 * create by guorg
 *
 * @flow
 */
export function createImageImport(images: Object[]): string {
  const result = [];
  if (!images) {
    return result.join(';');
  }
  images.forEach((item: Object) => {
    const { name, value } = item;
    result.push(`import ${name} from '${value}'`);
  });
  return result.join(';');
}
