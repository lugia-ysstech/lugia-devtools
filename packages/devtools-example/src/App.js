import React from 'react';
import Page from './pages/Page';

class App extends React.Component {
  render() {
    return <Page />;
  }
}

export default App;
