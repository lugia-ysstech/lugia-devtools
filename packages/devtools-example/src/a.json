{
  "type": "PolarBear",
  "designPoint": [-1660, -1940],
  "scale": 0.6,
  "areaId2ChildPadIndexString": {},
  "widgetId2ChildPad": {},
  "forkDependencies": [],
  "mainDependencies": [
    {
      "version": "1.0.0",
      "module": "@lugia/lugia-web",
      "widgetName": "Avatar"
    }
  ],
  "forkPad": {
    "point": [-1660, -1940],
    "width": 5000,
    "index": 0,
    "height": 5000,
    "zIndex": 4000,
    "type": "ForkPad",
    "layers": [],
    "idSeq": 67,
    "id2WidgetInfo": {},
    "id": "forkPad",
    "lastIndex": 41,
    "group": false
  },
  "mainPad": {
    "point": [-120, 20],
    "width": 1920,
    "index": 0,
    "height": 1080,
    "zIndex": 4000,
    "type": "MainPad",
    "layers": [
      {
        "rulerIndex": 5,
        "i": "wb(Aqu71",
        "z": 0,
        "w": 640,
        "h": 427,
        "p": [533, 436]
      }
    ],
    "idSeq": 91291,
    "id2WidgetInfo": {
      "wb(Aqu71": {
        "w": "Avatar",
        "a": "AvatarSrc",
        "m": "@lugia/lugia-web",
        "v": "1.0.0",
        "p": {
          "icon": "lugia-icon-financial_user",
          "name": "Lugia",
          "shape": "circle",
          "size": "default",
          "src": "9f2f070828381f30cfb1ff8e74179c0c6e06f00c.jpeg",
          "type": "img"
        },
        "i": "wb(Aqu71"
      }
    },
    "id": "mainPad",
    "lastIndex": 71,
    "group": false,
    "backgroudColor": "#f8f8f8"
  },
  "themes": {
    "widgetId2ThemeInfo": {}
  },
  "lugiax": {
    "page2mutation": {},
    "widgetId2PropsName2BindInfo": {},
    "widgetId2EventName2MutationInfo": {}
  },
  "assets": {
    "widgetIdHasAssetPropsName": {
      "wb(Aqu71": {
        "src": true
      }
    },
    "widgetIdHasAssetThemeName": {}
  },
  "widgetId2Locked": {},
  "layoutInfos": {
    "mode2Config": {
      "1920x1080": {
        "forkPadPoint": [-1660, -1940],
        "mainPadSize": {
          "height": 1080,
          "width": 1920
        },
        "scale": 0.6,
        "widthRange": [0, 50000]
      }
    },
    "mode2LayoutData": {
      "1920x1080": {
        "wb(Aqu71": {
          "theme": {},
          "a": "mainPad",
          "w": 640,
          "h": 427,
          "p": [533, 436],
          "wi": "wb(Aqu71"
        }
      }
    }
  },
  "zip": true
}
