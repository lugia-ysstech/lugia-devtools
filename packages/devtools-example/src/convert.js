/**
 *
 * create by ligx
 *
 * @flow
 */
const Convert = require('../../devtools-conversion/lib');
const Test = require('./a.json');
const fs = require('fs');

console.info(Convert);
fs.writeFileSync('./Page.js', Convert(Test, { resourcesHead: 'http:/' }));
