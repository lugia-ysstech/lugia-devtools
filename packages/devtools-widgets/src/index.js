/**
 *
 * create by grg on 2020/6/30
 *
 * @flow
 */
const {
  createDesignInfo,
  createThemeMeta
} = require("@lugia/devtools-material");
const getEntryContent = require("./component.entry");
const {
  ensureFileSync,
  outputJsonSync,
  writeFileSync,
  pathExistsSync,
  readJsonSync,
  unlinkSync
} = require("fs-extra");
const { join } = require("path");
process.env.NODE_ENV = "production";
process.env.NODE_ENV = "production";
process.env.__FROM_TEST = true;
process.env.ESLINT = "none";
process.env.TSLINT = "none";
process.env.COMPRESS = "none";
const singleCompile = require("./compile");
const gulp = require("gulp");
const buildTask = require("./gulpfile");
const DesignFile = "designInfo.js";
const ThemeMetaFile = "themeMeta.json";

async function createDesignInfoFile (config) {
  const {
    targetPath,
    componentInvalid,
    hideInTollPanelComponents,
    themeInvalid,
    extendCode
  } = config;
  const designInfoStr = await createDesignInfo(targetPath, componentInvalid, {
    outFile: "string",
    hideInTollPanelComponents
  });
  const designInfoPath = join(targetPath, DesignFile);
  ensureFileSync(designInfoPath);
  writeFileSync(designInfoPath, designInfoStr + extendCode);
  await createThemeMeta({ targetPath, invalid: themeInvalid });
}

module.exports = async function compileComponent (param) {
  const {
    componentInvalid,
    hideInTollPanelComponents,
    themeInvalid,
    extendCode = ""
  } = param;

  const cwd = process.cwd();
  const outputDir = join(cwd, "./src/widgets");
  const entry = "./src/widgets/component.entry.js";
  const entryFile = join(cwd, entry);
  const { name: moduleName, version } = readJsonSync(join(cwd, "package.json"));
  writeFileSync(entryFile, getEntryContent(moduleName, version));
  await createDesignInfoFile({
    targetPath: outputDir,
    componentInvalid,
    hideInTollPanelComponents,
    themeInvalid,
    extendCode
  });

  const { externals = {}, importModules } = param;
  const targetName = `__${moduleName}__${version}`;
  const config = {
    SINGLE_COMPILED: true,
    cwd,
    externals: {
      react: "React",
      "react-dom": "ReactDom",
      "@lugia/lugiax": "lugiax",
      "@lugia/theme-config": "lugiatheme",
      "@lugia/theme-core": "lugiadthemecore",
      "@lugia/dict": "lugiadict",
      "styled-components": "styled",
      ...externals
    },
    entry,
    name: targetName,
    outputDir,
    importModules,
    publicPath: `file:///${outputDir.replace(/\\/g, "\\\\")}/`,
    disableCssExtract: true
  };

  singleCompile(config, (error, result) => {
    if (!result) {
      return;
    }

    const { assets } = result;

    const { outputDir } = config;
    const getFullOutPath = target => join(outputDir, target);
    const dllFilePath = getFullOutPath("./designInfo.dll.json");
    const lugiadConfigFile = join(outputDir, "lugiad.config.json");
    let lugiadConfig = {};
    if (pathExistsSync(lugiadConfigFile)) {
      lugiadConfig = readJsonSync(lugiadConfigFile);
      console.info(`读取组件扩展信息`, lugiadConfig);
    }
    assets.forEach(async asset => {
      const {
        content,
        originalAsset: { name }
      } = asset;

      if (/\.js$/.test(name)) {
        if (outputDir) {
          ensureFileSync(dllFilePath);
          outputJsonSync(dllFilePath, {
            version,
            targetName,
            asset: {
              ...asset,
              content: content.toString()
            },
            lugiadConfig
          });
          unlinkSync(entryFile);
          const build = buildTask(importModules);
          build(() => {
            unlinkSync(dllFilePath);
            unlinkSync(getFullOutPath(ThemeMetaFile));
          });
        }
      }
    });
  });
};
