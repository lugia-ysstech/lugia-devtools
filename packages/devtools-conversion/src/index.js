/**
 *
 * create by guorg
 *
 * @flow
 */
import { createHeader, getModelCode, getLugiadCoreCode, getResponsiveCode } from './header';
import { createImageImport } from './img';
import { createComponent, createLayerComponent } from './createClass';
import { unZip } from '@lugia/devtools-core';

function getPageMutation(pageMutation: Object): string {
  if (!pageMutation) {
    return '';
  }
  const eventNames = Object.keys(pageMutation).filter(
    (name: string): boolean =>
      name === 'componentDidMount' || name === 'componentWillUnmount'
  );
  const res = [];
  eventNames.forEach((name: string) => {
    const { modelName, mutationName } = pageMutation[name];
    res.push(`${name}(){
          ${modelName}.mutations.${mutationName}({eventName: '${name}'});
      }`);
  });
  return res.join('');
}

export default function conversion(page: Object, options: Object): string {
  let exportCode = '';
  if (!page) {
    return exportCode;
  }
  const widgetId2Component = {};
  const images = [];
  const { zip } = page;
  if (zip) {
    page = unZip(JSON.stringify(page));
  }
  const { resourcesHead = '' } = options || {};
  const { mainDependencies = [], layoutInfos = {} } = page;
  const { mode2Config, mode2LayoutData } = layoutInfos;
  const isResponsive = Object.keys(mode2Config || {}).length > 1;
  const { packages, styledComponentCode } = createHeader(
    mainDependencies,
    widgetId2Component
  );
  const {
    mainPad,
    widgetId2ChildPad,
    lugiax = {},
    themes = {},
    assets = {},
  } = page;
  const { lugiadCoreCode, lugiadFuncCode } = getLugiadCoreCode(lugiax);
  const { widgetIdHasAssetPropsName = {} } = assets;
  const { children, layers, id2WidgetInfo, width, zIndex, height } = mainPad;
  const modelCode = getModelCode(lugiax);
  const { rspPackagesCode, rspDeconstructionCode: responsiveCode } = getResponsiveCode(layoutInfos);
  const classCode = createComponent(
    children,
    widgetId2ChildPad,
    lugiax,
    themes,
    widgetId2Component,
    isResponsive,
    { resourcesHead, widgetIdHasAssetPropsName, images }
  );
  const { layerCode, layerBindCode } = createLayerComponent(
    layers,
    id2WidgetInfo,
    lugiax,
    themes,
    'MainPad',
    widgetId2Component,
    isResponsive,
    { resourcesHead, widgetIdHasAssetPropsName, images }
  );
  let imageCode = createImageImport(images);
  imageCode = imageCode ? imageCode + ';' : '';
  const mode2ConfigData = JSON.stringify(mode2Config);
  const mode2LayoutDatas = JSON.stringify(mode2LayoutData);
  const nomalCode = `<div style={{width: '${width}px',height: '${height}px',zIndex: '${zIndex}', position: 'relative'}}>${layerCode}</div>`;
  const contextCode = `<DesignResponsive mode2Config={${mode2ConfigData}} mode2LayoutData={${mode2LayoutDatas}} sideMenuWidth={this.props.sideMenuWidth}>
                <ResponsiveContext.Consumer>{
                    context => {
                        return ${nomalCode}
                    }
                }</ResponsiveContext.Consumer>        
            </DesignResponsive>`;
  const Code = isResponsive ? contextCode : nomalCode;
  exportCode = `${packages} ${lugiadCoreCode} ${imageCode} ${modelCode} ${rspPackagesCode} ${responsiveCode} ${lugiadFuncCode} ${styledComponentCode} ${classCode} ${layerBindCode} export default class Page extends React.Component{
  ${getPageMutation(lugiax.page2mutation)}
      render(){
        return (
            ${Code}
        );
      }  
  }`;
  return exportCode;
}
