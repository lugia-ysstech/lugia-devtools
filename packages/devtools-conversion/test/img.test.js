/**
 *
 * create by guorg
 *
 * @flow
 */
import { createImageImport } from '../src/img';
import conversion from '../src';

describe('conversion createImageImport ->', () => {
  it('error arg', () => {
    const spaceCode = createImageImport();
    expect(spaceCode).toBe('');
    const errCode = createImageImport(undefined);
    expect(errCode).toBe('');
    const errorCode = createImageImport(null);
    expect(errorCode).toBe('');
    const code = createImageImport([]);
    expect(code).toBe('');
  });
  it('createImageImport', () => {
    const images = [
      { name: 'img1', value: './image.img1.jpg' },
      { name: 'img2', value: './image.img2.jpg' },
      { name: 'img3', value: './image.img3.jpg' },
      { name: 'img4', value: './image.img4.jpg' },
      { name: 'img5', value: './image.img5.jpg' },
    ];
    expect(createImageImport(images)).toMatchSnapshot();
  });
});
