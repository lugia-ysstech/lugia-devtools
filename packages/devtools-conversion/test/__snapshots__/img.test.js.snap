// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversion createImageImport -> createImageImport 1`] = `"import img1 from './image.img1.jpg';import img2 from './image.img2.jpg';import img3 from './image.img3.jpg';import img4 from './image.img4.jpg';import img5 from './image.img5.jpg'"`;
